src/mongocxx/test/client_side_encryption.cpp

```diff
@@ -227,6 +231,20 @@ void _add_cse_opts(options::client_encryption* opts,
     opts->key_vault_namespace({"keyvault", "datakeys"});
 }
 
+options::client crypt_shared_opts() {
+    options::client opts;
+    const auto shared_lib_path = std::getenv("CRYPT_SHARED_LIB_PATH");
+    if (shared_lib_path) {
+        opts.auto_encryption_opts(
+            options::auto_encryption()
+                .kms_providers(_make_kms_doc())
+                .key_vault_namespace({"keyvault", "datakeys"})
+                .extra_options(make_document(kvp("cryptSharedLibPath", shared_lib_path),
+                                             kvp("cryptSharedLibRequired", true))));
```

Comment:
Document `cryptSharedLibPath` and `cryptSharedLibRequired` in the `extra_options` function: https://github.com/mongodb/mongo-cxx-driver/blob/56717a466117a39fdaabc213f894636b7ba77ae2/src/mongocxx/options/auto_encryption.hpp#L323-L345

The C driver documentation could be used as a reference: http://mongoc.org/libmongoc/current/mongoc_auto_encryption_opts_set_extra.html